### DISCUSSION TITLE
problem with getDefaultCamera(); from Node

### DISCUSSION THREAD

**Discussion by GamesStudioTwo** [axmolengine/axmol#1982] [General]
Hi all I got a problem with using
Director::getInstance()->getRunningScene()->getDefaultCamera();
I'm using it in a sub class for Camera but its just crashing I just want to remove and add a new one from a sub class for Camera
its fine is it in the main Scene()  I  just call getDefaultCamera()  not a problem. but I'm calling it from a sub class.
but its not finding it.  there's only 1 scene running is it a bug.? or is there a new way to call from the Director
I have even tired getting Parent as well. 

I have even just tried
 auto cam = Director::getInstance()->getRunningScene()->getCameras();
noting else and it just crash out I'm not even asking to do anything.

**Comment by rh101** [comment]
Can you please provide a minimal test project that reproduces the issue?

Also, have you tried following it with a debugger to see specifically where and why it crashes?

**Reply by GamesStudioTwo** [reply]
its ok My fault was late last night just woke , I forgot we have an onEnter()
I was trying to call it from the Init() function.
I don't think I have ever needed or used an on Enter function before.

**Comment by GamesStudioTwo** [comment]
Damm. Got another problem now if I call it in OnEnter();
it finds it but I cannot remove it.
removeChild(getDefaultCamera());

**Reply by smilediver** [reply]
`Scene` doesn't support changing default camera. The only way to change it is to inherit `Scene` and in your class constructor set `_defaultCamera`. Just see `Scene.cpp` and search for `_defaultCamera` to see how it works. But I would avoid it, I don't know what other subtle things might depend on current setup. If you want to add more functionality to it, then you can do this in a custom "camera controller" node, that would simply work with the existing default camera.

**Comment by GamesStudioTwo** [comment]
Hi thanks for replay cant see anything in there 
I done a quick video of what's happing. its not a big problem.

https://www.youtube.com/watch?v=l3qNeQ5IAO4
